tuf/api/metadata.py

```diff
@@ -1274,12 +1274,14 @@ class DelegatedRole(Role):
     instance attributes.*
 
     Args:
-        name: Delegated role name.
```

Comment:
Yea, you are correct. We should start thinking as a real library and be careful with such changes.
Because of Python limitations, I had to move the `name` argument if it has a default value after all attributes without a default value.
I think maybe it will be better to require a value for `name` even if it's `None` and not move it.
Will try with that.